web/frontend/src/pages/form/IndividualResult.tsx

```diff
@@ -82,37 +117,172 @@ const IndividualResult: FC<IndividualResultProps> = ({
       </div>
     );
   };
+
+  const getResultData = (
+    subject: Subject,
+    dataToDownload: DownloadedResults[],
+    BallotID: number
+  ) => {
+    dataToDownload.push({ Title: subject.Title });
+
+    subject.Order.forEach((id: ID) => {
+      const element = subject.Elements.get(id);
+      let res = undefined;
+
+      switch (element.Type) {
+        case RANK:
+          const rankQues = element as RankQuestion;
+
+          if (rankResult.has(id)) {
+            res = rankResult.get(id)[BallotID].map((rank, index) => {
+              return {
+                Rank: `${index + 1}`,
+                Choice: rankQues.Choices[rankResult.get(id)[BallotID].indexOf(index)],
+              };
+            });
+            dataToDownload.push({ Title: element.Title, Results: res });
+          }
+          break;
+
+        case SELECT:
+          const selectQues = element as SelectQuestion;
+
+          if (selectResult.has(id)) {
+            res = selectResult.get(id)[BallotID].map((select, index) => {
+              const checked = select ? 'True' : 'False';
+              return { Candidate: selectQues.Choices[index], Checked: checked };
+            });
+            dataToDownload.push({ Title: element.Title, Results: res });
+          }
+          break;
+
+        case SUBJECT:
+          getResultData(element as Subject, dataToDownload, BallotID);
+          break;
+
+        case TEXT:
+          const textQues = element as TextQuestion;
+
+          if (textResult.has(id)) {
+            res = textResult.get(id)[BallotID].map((text, index) => {
+              return { Field: textQues.Choices[index], Answer: text };
+            });
+            dataToDownload.push({ Title: element.Title, Results: res });
+          }
+          break;
+      }
+    });
+  };
+
+  const exportJSONData = () => {
+    const fileName = `result_${configuration.MainTitle.replace(/[^a-zA-Z0-9]/g, '_').slice(
+      0,
+      99
+    )}_byBallot`;
```

Comment:
I would use the term "grouped", and "individual" instead of "byBallot" and "byQuestion". Also, I would use a double "__" so you have something like "My_great_form__grouped".

Reply:
great idea, done.